flocker/apiclient/_client.py

```diff
@@ -0,0 +1,90 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Client for the Flocker REST API.
+"""
+
+from uuid import UUID
+
+from zope.interface import Interface
+
+from pyrsistent import PClass, field, pmap_field
+
+
+class Dataset(PClass):
```

Comment:
:question: ``PClass`` sounds like it's probably more appropriate than ``PRecord`` for ``Dataset`` and ``DatasetState``. Ought we to be using ``PClass`` instead of ``PRecord`` for ``flocker.control._model.Dataset`` etc?